tests/integration/test_akafka_testutils.py

```diff
@@ -407,3 +450,75 @@ async def test_expect_events_mismatch(
 
     with pytest.raises(ValidationError):
         await event_recorder.__aexit__(None, None, None)
+
+
+@pytest.mark.parametrize(
+    "recorded_correlation_id", [DEFAULT_CORRELATION_ID, OTHER_CORRELATION_ID]
+)
+@pytest.mark.parametrize(
+    "expected_headers",
+    [
+        None,
+        {"correlation_id": OTHER_CORRELATION_ID},
+        {"correlation_id": DEFAULT_CORRELATION_ID},
+    ],
+)
+@pytest.mark.parametrize("capture_headers", [True, False])
+async def test_capture_headers(
+    expected_headers: Optional[dict[str, str]],
+    recorded_correlation_id: str,
+    capture_headers: bool,
+    kafka: KafkaFixture,
+):
+    """Test the handling of mismatches between recorded and expected events using the
+    expect_events method of the KafkaFixture, but specifically with the optional headers.
+    For now, the only header to be tested is the correlation_id.
+
+    The expected behavior is that the headers are compared when provided as expected
+    arguments, otherwise ignored.
+    """
+    topic = "test_topic"
+
+    config = KafkaConfig(
+        service_name="test_publisher",
+        service_instance_id="1",
+        kafka_servers=kafka.kafka_servers,
+    )
+    expected_event = ExpectedEvent(
+        payload={"test_content": "Hello"},
+        type_="test_type",
+        headers=expected_headers,
+    )
+
+    # Set up the event recorder with the expected event and the capture_headers flag
+    event_recorder = kafka.expect_events(
+        events=[expected_event], in_topic=topic, capture_headers=capture_headers
+    )
+
+    # start the recorder
+    await event_recorder.__aenter__()
+
+    # Set the correlation ID context var and publish the 'recorded' event
+    async with (
+        set_correlation_id(recorded_correlation_id),
+        KafkaEventPublisher.construct(config=config) as event_publisher,
+    ):
+        await event_publisher.publish(
+            payload=expected_event.payload,
+            type_=expected_event.type_,
+            key="",
+            topic=topic,
+        )
+
+    # only time there should NOT be an error is:
+    # 1. Expected headers are not provided
+    # 2. Capture headers flag is set and expected/recorded headers match
+    error = True
+    if not expected_headers or (
+        capture_headers
+        and (expected_headers.get("correlation_id", "") == recorded_correlation_id)
+    ):
+        error = False
+
+    with pytest.raises(ValidationError) if error else nullcontext():
```

Comment:
```suggestion
    # only time there should be an error is when 1. expected headers are provided
    # and 2. capture headers flag aren't set or expected/recorded headers don't match
    expected_error = expected_headers and (
        not capture_headers
        or expected_headers.get("correlation_id", "") != recorded_correlation_id
    )

    with pytest.raises(ValidationError) if expected_error else nullcontext():
```

Reply:
Thanks, this is a little easier to read. I added this and the previous changes.